ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+
+In this area the differences between Akka and Orleans are very pronounced and mirror the different intent stated up-front: while Orleans aims for convenience and uses syntax that is familiar to OOP practitioners, Akka makes messaging very explicit and requires the user to define message classes. There is one instance of providing a similar model as async–await, namely the interaction between a PersistentActor and its Journal using the `persist` and `persistAsync` methods, but this API leads to a stronger coupling of the two components than would otherwise be the case. For a PersistentActor this is intended since it cannot function without the Journal, but for Grain interactions users now have to choose between blocking the grain in an `await` call (in a logical sense, but still!) or permitting other invocations to be processed concurrently “during” the `await` call—which leads to much of the same race conditions and problems that shared-state concurrency has. In Akka the interaction of multiple Actors is clearly demarcated by sending messages, and if and Actor ignores all incoming requests until a certain reply has been received then that is an explicit choice that the programmer makes.
+
+Another consequence of using interface methods for messaging is that all interactions default to the request–response pattern whereas explicit messaging naturally allows more streamlined message flows through multiple stations.
+
+#### Actor References
+
+  * Orleans Actor references do not contain any location information, all invocations will be resolved using the distributed directory when needed.
+  
+  * Akka ActorRef contains all information that is necessary in order to send a message to the Actor it designates, including interactions with Actors outside of the current cluster.
+
+#### Promises
+
+  * Orleans relies upon the async–await language feature to formulate continuations that are invoked asynchronously. This means that the user need only ensure that methods return Promises and can then write code as if it was synchronous.
+  
+  * Instead of trying to offer a familiar (synchronous) user experience Akka makes handling asynchronous results explicit. In the case of the `ask`-pattern—which is analogous to how every invocation works in Orleans—the user should transform the Future explicitly or send its value to an Actor by way of the `pipeTo` pattern.
+
+We found that making asynchronous code look synchronous invites many subtle possibilities of making mistakes, either concerning data consistency (i.e. transparently interleaving different invocations by way of their scheduled closures) or liveness (by blocking a Grain from processing any message until some response has been received). The nature of distributed systems is that messages are sent one-way and with at-most-once semantics, that is what physics gives us, and suggesting that there will always be a reply often turns out to be a pitfall, especially for unexperienced users.
```

Comment:
As I said before, this moves the problem of correlating request to response to a different place, presenting now a different difficulty.